src/js/components/FrameworkConfiguration.js

```diff
@@ -0,0 +1,387 @@
+import CosmosErrorMessage from "#SRC/js/components/CosmosErrorMessage";
+import FrameworkConfigurationForm
+  from "#SRC/js/components/FrameworkConfigurationForm";
+
+export default class FrameworkConfiguration extends Component {
+  constructor(props) {
+    super(props);
+
+    const schema = props.packageDetails.getConfig();
+    const activeTab = Object.keys(schema.properties)[0];
+    const focusField = Object.keys(schema.properties[activeTab].properties)[0];
+
+    this.state = {
+      reviewActive: props.isInitialDeploy,
+      activeTab,
+      focusField,
+      jsonEditorActive: false,
+      hasChangesApplied: false,
+      isConfirmOpen: false,
+      isOpen: true
+    };
+  }
+
+  onFormDataChange(formData) {
+    this.props.onFormDataChange(formData);
+
+    this.setState({ hasChangesApplied: true });
+  }
+
+  onFormErrorChange(formErrors) {
+    this.props.onFormErrorChange(formErrors);
+  }
+
+  onFocusFieldChange(activeTab, focusField) {
+    if (deepEqual(focusField, this.state.focusField)) {
+      return false;
+    }
+
+    this.setState({ focusField, activeTab });
+  }
+
+  onActiveTabChange(activeTab) {
+    const { currentActiveTab } = this.state;
+    const { packageDetails } = this.props;
+
+    if (deepEqual(activeTab, currentActiveTab)) {
+      return false;
+    }
+
+    const schema = packageDetails.getConfig();
+    const focusField = Object.keys(schema.properties[activeTab].properties)[0];
+
+    this.setState({ activeTab, focusField });
+  }
+
+  onReviewConfigurationRowClick(rowData) {
+    const { tabViewID } = rowData;
+
+    const keyPath = (tabViewID || "").split(".");
+    if (keyPath.length < 2) {
+      this.onEditConfigurationButtonClick();
+
+      return false;
+    }
+
+    const activeTab = keyPath[0];
+    const focusField = keyPath[1];
+
+    this.setState({ reviewActive: false, focusField, activeTab });
+  }
+
+  onEditConfigurationButtonClick() {
+    const { packageDetails } = this.props;
+
+    const schema = packageDetails.getConfig();
+    const activeTab = Object.keys(schema.properties)[0];
+    const focusField = Object.keys(schema.properties[activeTab].properties)[0];
+
+    this.setState({ reviewActive: false, activeTab, focusField });
+  }
+
+  getHashMapRenderKeys(formData, renderKeys) {
+    if (!Util.isObject(formData)) {
+      return false;
+    }
+
+    Object.keys(formData).forEach(key => {
+      const formattedKey = key
+        .toLowerCase()
+        .split("_")
+        .map(word => word[0].toUpperCase() + word.slice(1))
+        .join(" ");
+
+      renderKeys[key] = formattedKey;
+      this.getHashMapRenderKeys(formData[key], renderKeys);
+    });
+  }
+
+  handleJSONToggle() {
+    this.setState({ jsonEditorActive: !this.state.jsonEditorActive });
+  }
+
+  handleCloseConfirmModal() {
+    this.setState({ isConfirmOpen: false });
+  }
+
+  handleGoBack() {
+    const { reviewActive, hasChangesApplied } = this.state;
+
+    if (reviewActive && hasChangesApplied) {
```

Comment:
Personal taste I would prefer early exit as it will remove complexity
```JS
if (reviewActive && hasChangesApplied) {
   this.setState({ reviewActive: false });

   return;
}
if (hasChangesApplied) {
  this.setState({ isConfirmOpen: true });

  return;
}
this.handleConfirmGoBack();
```

Feel free to adjust.